python/ray/serve/_private/proxy_request_response.py

```diff
@@ -160,28 +161,30 @@ def user_request(self) -> bytes:
     def send_request_id(self, request_id: str):
         self.context.set_trailing_metadata([("request_id", request_id)])
 
-    def send_status_code(self, status_code: grpc.StatusCode):
-        self.context.set_code(status_code)
-
-    def send_details(self, message: str):
-        self.context.set_details(message)
-
     def request_object(self, proxy_handle: ActorHandle) -> gRPCRequest:
         return gRPCRequest(
             grpc_user_request=self.user_request,
             grpc_proxy_handle=proxy_handle,
         )
 
 
-class ProxyResponse:
-    """ProxyResponse class to use in the common interface among proxies"""
+@dataclass(frozen=True)
+class ResponseStatus:
+    code: Any  # Must be convertible to a string.
```

Comment:
Why not just making it a string?

Reply:
gRPC uses an integer enum

Reply:
and it's not trivial to convert from str -> enum value :(

Reply:
are you referring to https://grpc.github.io/grpc/python/_modules/grpc.html#StatusCode?

Reply:
Can we then do `Union[str, StatusCode]` instead of Any?